src/commands/createFunctionApp/FunctionAppCreateStep.ts

```diff
@@ -167,6 +228,37 @@ export class FunctionAppCreateStep extends AzureWizardExecuteStep<IFunctionAppWi
         newSiteConfig.appSettings = appSettings;
         return newSiteConfig;
     }
+
+    async createFunctionApp(context: IFunctionAppWizardContext, rgName: string, siteName: string, stack: FullFunctionAppStack): Promise<Site> {
+        const client: WebSiteManagementClient = await createWebSiteClient(context);
+        return await client.webApps.beginCreateOrUpdateAndWait(rgName, siteName, await this.getNewSite(context, stack));
+    }
+
+    async createFlexFunctionApp(context: IFunctionAppWizardContext, rgName: string, siteName: string): Promise<Site> {
+        const headers = createHttpHeaders({
+            'Content-Type': 'application/json',
+        });
+
+        const options: AzExtRequestPrepareOptions = {
+            url: `https://management.azure.com/subscriptions/${context.subscriptionId}/resourceGroups/${rgName}/providers/Microsoft.Web/sites/${siteName}?api-version=2023-12-01`,
+            method: 'PUT',
+            body: JSON.stringify(await this.getNewFlexSite(context)) as unknown as RequestBodyType,
+            headers
+        };
+
+        const client = await createGenericClient(context, context);
+        const result = await client.sendRequest(createPipelineRequest(options)) as AzExtPipelineResponse;
+        if (result) {
+            const client: WebSiteManagementClient = await createWebSiteClient(context);
+            // the payload for the new API version "2023-12-01" is incompatiable with our current SiteClient so get the old payload
+            try {
+                return await client.webApps.get(rgName, siteName);
```

Comment:
Are we pretty sure that by the time the `AzExtPipelineResponse` result is passed back, that the web app has finished building/allocating? Wondering if there will ever be a timing issue here